askama_parser/src/node.rs

```diff
@@ -93,9 +93,21 @@ impl<'a> Node<'a> {
         let (i, _) = s.nest(j)?;
         let result = func(i, s);
         s.leave();
-
         let (i, node) = result?;
-        let (i, _) = cut(|i| s.tag_block_end(i))(i)?;
+
+        let (i, closed) = cut(alt((
+            value(true, |i| s.tag_block_end(i)),
+            value(false, ws(eof)),
+        )))(i)?;
+        if !closed {
```

Comment:
Since it's always almost the same except for one word, better put it into a function returning `Result<(), nom::Err>`, no?

Reply:
Yes, makes sense to unify that into one helper function. Done.